### PR TITLE
Merges security into 

### PR DESCRIPTION
**Pull request by hoptond**
Unit tested, dockblocked, minimum viable product. Not much code to look through, one unit test.

Code is login.php,logout.php,security_functions.php

### CODE REVIEW

**Comment by robertkieronreeves** [comment]
Couldn't find any problems.

File: edit/login.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php require 'security_functions.php';
+
+session_start();
+if (verifyLogin($_POST['username'], $_POST['password'])) {
```
Comment (mporam) [inline edit/login.php:4]: what if username or password keys dont exist?

File: edit/login.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    setcookie('id', session_id());
```
Comment (mporam) [inline edit/login.php:5]: you dont need to do this, session_start() does it for you

File: edit/logout.php [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?php require 'security_functions.php';
+
+setcookie('id',0, time() - 1);
```
Comment (mporam) [inline edit/logout.php:3]: use `session_destroy()` instead, and dont do any cookie stuff

File: edit/security_functions.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    return false;
+}
+
+/*
+ * If the cookie ID does not match the session ID, the user is punted out of the session. GET OUTTA HERE
+ */
+function verifyUser() {
+    session_start();
+    if ($_COOKIE['id'] != session_id()) {
```
Comment (mporam) [inline edit/security_functions.php:28]: this is not needed, just check a key in the session

Comment (mporam) [reply edit/security_functions.php:28]: doesnt actually add any real security for session id stealing as its easy to fake a cookie

File: edit/security_functions.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ * Verifies the username and password. If it is correct, any session data relating to incorrect entry is removed.
+ *
+ * @return string Returns a message if either the username or password is incorrect. If the user has entered no data,
+ * this value is blank.
+ */
+function loginFailMessage() : string {
+    session_start();
+    if (array_key_exists('badname' , $_SESSION)) {
+        session_destroy();
+        return 'User not in database. Please try again';
```
Comment (mporam) [inline edit/security_functions.php:45]: by telling the user if they have a incorrect user or password you are opening up yourself to a brute force attack, people can just keep trying usernames until the error doesnt appear. should show a generic error that says incorrect username or password. poorer UX but better security

**Review by mporam — CHANGES_REQUESTED** [review]

**Comment by hoptond** [comment]
project week over, closing this PR for the moment

**Pull request closed without merging**